### DISCUSSION TITLE
Transaction insights are changing. Here's what that means for your Snap.

### DISCUSSION THREAD

**Discussion by Montoya** [MetaMask/snaps#2301] [Announcements]
We will soon be rolling out a new version of transaction insights that introduces a different approach to how insights are fetched in MetaMask. In the current implementation, an insight is fetched when the user clicks on the insight tab to view the insight. Example: 

https://github.com/MetaMask/snaps/assets/19909/9dd9d8d1-1d39-4178-9d9e-a4b9cb360a98

Note that the insight loads after the tab is activated. 

In a future update, transaction insights will be fetched as soon as the transaction is requested. This is necessary for Snaps to be able to return a [severity flag](https://docs.metamask.io/snaps/reference/entry-points/#transaction-severity-level), which warns the user before they confirm a potentially dangerous transaction. 

What this means for transaction insight Snaps is that the Snap will be loaded on every transaction request, even if the insight is not viewed by the user. You may need to update your Snap logic as a result to account for this. If you have questions, please reply in this thread and we can support you on next steps.

**Comment by mindofmar** [comment]
Awesome!

1. Keep us posted on what the new API/endowment will look like.
2. What will the UI look like when it is `CRITICAL`?
3. Can we expect any other severities other than `CRITICAL`? It would be really nice to have another level like `HIGH` or `WARN` for lesser severity heuristics. (ie: `CRITICAL` = red confirmation button, whereas `WARN` maybe just shows a banner but still encourages a user to review.)
